JS book list: new entry + formatting for consistency

CristinaMendonca opened the pull request:
Worked on the JavaScript section:
- Capitalized the titles consistently;
- Added info of authors and book types for elements of the JS book list that had that info missing;
- Removed Learned Code JavaScript by Playing a Game (not a book);
- Removed the JS Books meta-list, as it's already in the correct section (meta-list section);
- Added the book "Designing Scalable JavaScript Applications" and "Thinking in JavaScript".

## What does this PR do?
Add Resource(s) | Improve Repo

### Description 
### Why is this valuable (or not)? 
Book on design patterns and programming JS with scalability in mind + introductory book in JS.
### How do we know it's really free? 
The publisher's site lists it as free + offered as free in Amazon.com (permanently, not a sale)
### For book lists, is it a book? For course lists, is it a course? 
It's a book (pdf format + kindle)

### Checklist:
- [X] Not a duplicate
- [X] Included author(s) if appropriate
- [X] Lists are in alphabetical order
- [X] Needed indications added (PDF, access notes, under construction)

SethiPandi commented:
Hey, and welcome to free programming books! :wave:

The PDF that's no longer free, it might be nice to remove the link entirely instead.

I've noticed you changed the titles of many resources so that their casing no longer matches the source.
I completely appreciate that for a title, it's much cleaner to use title case, but we prefer to use the sources title as is without modification, including casing.

For example, for:
https://2ality.com/2013/06/basic-javascript.html

The title is: "Basic JavaScript for the impatient programmer" not "Basic JavaScript for the Impatient Programmer".

CristinaMendonca commented:
> 
> 
> Hey, and welcome to free programming books! 👋
> 
> The PDF that's no longer free, it might be nice to remove the link entirely instead.
> 
> I've noticed you changed the titles of many resources so that their casing no longer matches the source.
> I completely appreciate that for a title, it's much cleaner to use title case, but we prefer to use the sources title as is without modification, including casing.
> 
> For example, for:
> https://2ality.com/2013/06/basic-javascript.html
> 
> The title is: "Basic JavaScript for the impatient programmer" not "Basic JavaScript for the Impatient Programmer".

Ok, I removed the book that is no longer free and reverted the titles. Thanks!

eshellman commented:
Thanks, and welcome to free-programming-books!
